test/e2e/tests/hardware-wallets/ledger/ledger-swap.spec.ts

```diff
@@ -0,0 +1,88 @@
+import { Browser } from 'selenium-webdriver';
+import { checkActivityTransaction } from '../../swaps/shared';
+import SettingsPage from '../../../page-objects/pages/settings/settings-page';
+import FixtureBuilder from '../../../fixture-builder';
+import { WINDOW_TITLES, withFixtures } from '../../../helpers';
+import { loginWithoutBalanceValidation } from '../../../page-objects/flows/login.flow';
+import HeaderNavbar from '../../../page-objects/pages/header-navbar';
+
+import HomePage from '../../../page-objects/pages/home/homepage';
+import AdvancedSettings from '../../../page-objects/pages/settings/advanced-settings';
+import SwapPage from '../../../page-objects/pages/swap/swap-page';
+
+import { mockLedgerTransactionRequests } from './mocks';
+
+const isFirefox = process.env.SELENIUM_BROWSER === Browser.FIREFOX;
+
+describe('Ledger Swap', function () {
+  it('swaps ETH to DAI', async function () {
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder().withLedgerAccount().build(),
+        localNodeOptions: {
+          hardfork: 'london',
+          loadState: './test/e2e/seeder/network-states/with50Dai.json',
+        },
+        title: this.test?.fullTitle(),
+        testSpecificMock: mockLedgerTransactionRequests,
+      },
+      async ({ driver }) => {
+        await loginWithoutBalanceValidation(driver);
+        const homePage = new HomePage(driver);
+        await homePage.checkExpectedTokenBalanceIsDisplayed('20', 'ETH');
+
+        // disable smart transactions
+        const headerNavbar = new HeaderNavbar(driver);
+        await headerNavbar.checkPageIsLoaded();
+        await headerNavbar.openSettingsPage();
+        const settingsPage = new SettingsPage(driver);
+
+        await settingsPage.checkPageIsLoaded();
+        await settingsPage.clickAdvancedTab();
+        const advancedSettingsPage = new AdvancedSettings(driver);
+        await advancedSettingsPage.checkPageIsLoaded();
+        await advancedSettingsPage.toggleSmartTransactions();
+        await settingsPage.closeSettingsPage();
+
+        await homePage.checkIfSwapButtonIsClickable();
+
+        await homePage.startSwapFlow();
+
+        await driver.delay(5000);
```

Comment:
typically we avoid hardcoded delays, and in case we cannot avoid them, it's good to add a comment on top, explaining why do we need the delay as well as the window switch below -- which is due to the application re-loading/opening  a new page, because of the ledger selected account + mock)